components/constellation/constellation.rs

```diff
@@ -1405,10 +1405,6 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
 
             let window_size = old_pipeline.and_then(|old_pipeline| old_pipeline.size);
 
-            if let Some(old_pipeline) = old_pipeline {
```

Comment:
There is explicit text about clearing timers when unloading a document: https://html.spec.whatwg.org/multipage/browsers.html#unloading-document-cleanup-steps step 3.2, invoked from step 10 of https://html.spec.whatwg.org/multipage/browsers.html#unload-a-document.